app/models/token.rb

```diff
@@ -51,26 +51,15 @@ def jwt_string
 
   # Decode the json web token when we receive it from the client
   def token
-    @token ||= JWT.decode(jwt_string, secret_key, true, verify_expiration: false)
+    @token ||= JWT.decode(jwt_string, secret_key, true, algorithm: hmac_sha_algo)
   end
 
   # Is the JWT token authentic?
   # - exp should be in the future
   # - iat should be in the past
   def token_validation
-    if (!jwt_string.blank? && !(token.all? &:blank?))
-      cur_time = Time.now
-      iat_time = Time.at(token[0]["iat"])
-      exp_time = Time.at(token[0]["exp"])
-      if exp_time < cur_time
-        errors.add(:base, I18n.t('token.expired'))
-      elsif !(iat_time < cur_time && iat_time < exp_time)
-        errors.add(:base, I18n.t('token.invalid'))
-      end
-    else
-      errors.add(:base, I18n.t('token.invalid'))
-    end
-  rescue JWT::DecodeError
+    raise JWT::DecodeError unless !jwt_string.blank? && !(token.all? &:blank?)
+  rescue JWT::DecodeError, JWT::ExpiredSignature
```

Comment:
why have the token validation methods been removed?

Reply:
cc @patrixr @bharat619

Reply:
Yeah I think that we should restore this file.

I think generally we don't want non-rails upgrade related changes in this PR. It makes it harder to review and will cause conflicts. Don't think random refactoring is a good on such a huge scale.

Reply:
I am not sure if earlier used version of jwt did validation on its own for `iat` or `exp`. So I just removed that, since there was some error that was thrown in this part.
I dont exactly remember what was it though

Reply:
Great, I want to double check this but if JWT is now doing it's own validation on decode then I'm all for it

Reply:
@steveyken 
```
2.5.1 :021 > exp = 1.day.ago.to_i
 => 1600148425
2.5.1 :022 > exp_payload = { data: 'data', exp: exp }
 => {:data=>"data", :exp=>1600148425}
2.5.1 :023 > token = JWT.encode exp_payload, hmac_secret, 'HS256'
 => "eyJhbGciOiJIUzI1NiJ9.eyJkYXRhIjoiZGF0YSIsImV4cCI6MTYwMDE0ODQyNX0.7KVt8FGaR-tuzAEr-X4YUoYxLYH9XFXdcqdP3Q_P9ac"
2.5.1 :024 > JWT.decode token, hmac_secret, true, { algorithm: 'HS256' }
Traceback (most recent call last):
        1: from (irb):24
JWT::ExpiredSignature (Signature has expired)
2.5.1 :025 >
```